mlrun/api/api/endpoints/feature_sets.py

```diff
@@ -32,16 +32,39 @@ def create_feature_set(
     )
 
 
-@router.put("/projects/{project}/feature_sets/{name}/references/{reference}")
-def update_feature_set(
+@router.put(
+    "/projects/{project}/feature_sets/{name}/references/{reference}",
+    response_model=schemas.FeatureSet,
+)
+def store_feature_set(
+    project: str,
+    name: str,
+    feature_set: schemas.FeatureSet,
+    reference: str,
+    versioned: bool = True,
+    db_session: Session = Depends(deps.get_db_session),
+):
+    tag, uid = parse_reference(reference)
+    uid = get_db().store_feature_set(
+        db_session, project, name, feature_set, tag, uid, versioned
+    )
+
+    return get_db().get_feature_set(db_session, project, name, uid=uid,)
+
+
+@router.patch("/projects/{project}/feature_sets/{name}/references/{reference}")
+def patch_feature_set(
     project: str,
     name: str,
-    feature_set_update: schemas.FeatureSetUpdate,
+    feature_set_update: dict,
```

Comment:
Why `dict` and not some schema ?

Reply:
Because this is a `patch` operation, there are no mandatory fields in the schema, and therefore defining a schema here is redundant. That's why I'm just accepting a dict that can contain anything in it. The alternative was to have an empty schema with `extra` in its configuration, but that doesn't make sense.

Reply:
I see what you're saying, I would generally prefer to do have a schema, but since I do not want to have a duplicate of everything, and can't find an easy way to "take this object definition but make everything optional recursively" (the most close one is this https://stackoverflow.com/questions/61948723/how-to-extend-a-pydantic-object-and-change-some-fileds-type, see woody's answer, but this feels too hacky, and also I don't think it's recursive)
I'm fine with leaving it a dict, hopefully there will be a better solution for it pydantic-wise, without hacks in the future